### PR TITLE
update for 4.18

### PR DESCRIPTION
**Pull request by odoe**
This PR updates the amd and webpack samples to latest 4.18. It also adds a RequireJS sample.

### CODE REVIEW

File: 4.x/amd/dojo/build.profile.js [hunk 1]
```diff
@@ -60,21 +61,7 @@ var profile = {
...
-      main: "moment",
-      packageJson: {},
-      resourceTags: {
-        miniExclude: function (filename, mid) {
-          if (mid.indexOf("/min/") > -1 || mid.indexOf("/src/") > -1) {
-            return true;
-          }
-          return ["moment/ender", "moment/package"].indexOf(mid) > -1;
-        },
-      },
```
Comment (dasa) [inline 4.x/amd/dojo/build.profile.js:77]: Can remove any other packages that the app is not using now that the package has no dependencies.

File: 4.x/amd/requirejs/README.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# npm Sample application
+
+This is a sample application showing how to use [npm](https://www.npmjs.com/) to build your [ArcGIS API for JavaScript](https://developers.arcgis.com/javascript/) application.
+It provides multiple types of ways to run your local builds using RequireJS.
+
+It provides a [Gruntfile](Gruntfile.js).
+
+# Requirements
+* [node & npm](https://nodejs.org/)
+* [java 7 or greater](https://java.com/en/download/) - for [Closure Compiler](https://github.com/google/closure-compiler) used during build
```
Comment (dasa) [inline 4.x/amd/requirejs/README.md:10]: Is this using Closure?

File: 4.x/amd/requirejs/src/index.html [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        addPkgIfNotPresent({
+          name: "maquette-jsx",
+          location: "../maquette-jsx",
+          main: "dist/maquette-jsx.umd"
+        });
+        addPkgIfNotPresent({
+          name: "tslib",
+          location: "../tslib",
+          main: "tslib"
+        });
```
Comment (dasa) [inline 4.x/amd/requirejs/src/index.html:127]: Can also remove unused packages.

File: 4.x/webpack/demo/package.json [hunk 4]
```diff
@@ -2,38 +2,70 @@
...
+			"plugin:@typescript-eslint/recommended",
+			"prettier/@typescript-eslint",
+			"plugin:prettier/recommended"
+		]
+	},
+	"prettier": {
+		"singleQuote": true,
+		"tabWidth": 4,
+		"useTabs": true
+	},
```
Comment (dasa) [inline 4.x/webpack/demo/package.json:64]: tabs -> spaces

File: 4.x/webpack/demo/src/index.ts [hunk 5]
```diff
@@ -1,117 +1,72 @@
...
+	});
+	//clear the filters when user closes the expand widget
+	seasonsExpand.watch('expanded', function () {
+		if (!seasonsExpand.expanded) {
+			// set it to an empty filter
+			floodLayerView.filter = new FeatureFilter();
+		}
+	});
+	view.ui.add(seasonsExpand, 'top-left');
+	view.ui.add('titleDiv', 'top-right');
```
Comment (dasa) [inline 4.x/webpack/demo/src/index.ts:71]: tabs

**Review by dasa — CHANGES_REQUESTED** [review]

File: 4.x/amd/requirejs/package.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "build": "grunt build"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "browserslist": "> 0.25%, not dead",
```
Comment (andygup) [inline 4.x/amd/requirejs/package.json:12]: I wasn't sure where to verify this. Is `browserslist` setting still correct?

File: 4.x/webpack/demo/src/css/main.css [hunk 7]
```diff
@@ -0,0 +1,2 @@
+@import "~arcgis-js-api/assets/esri/themes/light/main.css";
```
Comment (andygup) [inline 4.x/webpack/demo/src/css/main.css:1]: You used the dark css theme in the dojo example, maybe make this dark for consistency?

File: 4.x/amd/README.md [hunk 8]
```diff
@@ -1,6 +1,6 @@
 # arcgis-js-api build tools
 
-These are the recommended dojo build tools for using the [arcgis-js-api](https://github.com/Esri/arcgis-js-api) using npm.
+These are the recommended dojo and requirejs build configurations for using the [arcgis-js-api](https://github.com/Esri/arcgis-js-api) using npm.
```
Comment (andygup) [inline 4.x/amd/README.md:3]: Minor consistency issue with capitalization - RequireJS

**Review by andygup — CHANGES_REQUESTED** [review]

**Review by andygup — APPROVED** [review]
LGTM.

**Comment by odoe** [comment]
The Dojo and RequireJS samples have been updated.

* Updated the Dojo build file
* Update Dojo sample to run unbuilt
* Update RequireJS sample to run unbuilt

File: 4.x/amd/dojo/package.json [hunk 9]
```diff
@@ -28,7 +28,7 @@
   },
   "dependencies": {
     "arcgis-js-api": "~4.18.0",
-    "dojo": "~1.16.3",
+    "dojo": "^1.16.3",
     "whatwg-fetch": "^3.0.0"
```
Comment (dasa) [inline 4.x/amd/dojo/package.json:32]: Remove from here and from build.profile

File: 4.x/amd/dojo/src/index.html [hunk 10]
```diff
@@ -87,24 +87,6 @@
...
-          location: "../cldrjs",
-          main: "dist/cldr"
-        });
-        addPkgIfNotPresent({
-          name: "globalize",
-          location: "../globalize",
-          main: "dist/globalize"
-        });
         addPkgIfNotPresent({
           name: "maquette",
```
Comment (dasa) [inline 4.x/amd/dojo/src/index.html:91]: can be removed

File: 4.x/amd/dojo/build.profile.js [hunk 11]
```diff
@@ -1,734 +1,545 @@
...
+        trees: [
+          // don"t bother with .hidden, tests, min, src, and templates
+          [".", ".", /(\/\.)|(~$)|(node_modules)/],
+        ]
       },
       {
-        name: "moment",
-        location: "../moment",
-        main: "moment",
+        name: "maquette-jsx",
```
Comment (dasa) [inline 4.x/amd/dojo/build.profile.js:378]: can be removed

File: 4.x/amd/dojo/build.profile.js [hunk 12]
```diff
@@ -1,734 +1,545 @@
...
       },
-    ],
-    map: {
-      globalize: {
-        cldr: "cldrjs/dist/cldr",
-        "cldr/event": "cldrjs/dist/cldr/event",
-        "cldr/supplemental": "cldrjs/dist/cldr/supplemental",
-        "cldr/unresolved": "cldrjs/dist/cldr/unresolved",
+      {
+        name: "maquette-css-transitions",
```
Comment (dasa) [inline 4.x/amd/dojo/build.profile.js:390]: can be removed

File: 4.x/amd/dojo/build.profile.js [hunk 13]
```diff
@@ -1,734 +1,545 @@
...
+            return (
+              mid.indexOf("/tslib.es6") > -1 ||
+              mid.indexOf("/tslib.html") > -1 ||
+              mid.indexOf("/docs/") > -1
+            );
+          },
+        },
+      },
+      {
+        name: "whatwg-fetch",
```
Comment (dasa) [inline 4.x/amd/dojo/build.profile.js:416]: can be removed

File: 4.x/amd/dojo/build.profile.js [hunk 14]
```diff
@@ -0,0 +1,545 @@
...
+        resourceTags: {
+          miniExclude: function (filename, mid) {
+            return (
+              mid.indexOf("/dist/") > -1 && filename.indexOf(".umd.js") === -1
+            );
+          },
+        },
+      },
+      {
+        name: "tslib",
```
Comment (dasa) [inline 4.x/amd/dojo/build.profile.js:402]: can be removed

File: 4.x/amd/dojo/package.json [hunk 15]
```diff
@@ -23,12 +23,12 @@
     "gulp": "^4.0.2",
     "gulp-clean": "^0.4.0",
     "gulp-rename": "^1.2.2",
-    "node-sass": "~4.14.1",
+    "sass": "^1.30.0",
     "uglify-js": "^3.6.0"
```
Comment (dasa) [inline 4.x/amd/dojo/package.json:27]: Can this be removed?

File: 4.x/amd/dojo/package.json [hunk 16]
```diff
@@ -13,7 +13,7 @@
   },
   "license": "Apache-2.0",
   "devDependencies": {
-    "dojo-util": "github:dojo/util",
+    "dojo-util": "github:dojo/util#1.16.3",
```
Comment (dasa) [inline 4.x/amd/dojo/package.json:16]: Can this be same as `dojo`? `^1.16.3`

**Review by dasa — CHANGES_REQUESTED** [review]

File: 4.x/amd/dojo/package.json (hunk 15 above)
Comment (odoe) [reply 4.x/amd/dojo/package.json:27]: We still have a sass build guide, but I can get Franco to review that to make sure it's up to date

**Comment by odoe** [comment]
This got merged a little early, I'll make another PR with fixes

File: 4.x/amd/dojo/package.json (hunk 15 above)
Comment (dasa) [reply 4.x/amd/dojo/package.json:27]: Sorry, I meant can uglify-js be removed?

**Pull request merged**